### ISSUE TITLE
Only test accuracy does not increase, even Train accuracy is nealry 1

### ISSUE THREAD

**Issue by matrlot** [BVLC/caffe#3364]
I am using CNN which is based on mnist/lenet_train_test.prototxt and solver.prototxt
I changed only a few parameters to fit my data.

I tries to train but as you can see from the log below,
test accuracy does not increase more than 0.1

Does anyone have same problem?

I already checked #401 and older post
and that says problem are
1) old GPU driver
2) weights initialization(initial bias should be below than 0.7 or 0.5)
I already fixed these 2 things but nothing changed.

I1120 10:36:44.920687  6493 solver.cpp:340] Iteration 2880, Testing net (#0)
I1120 10:36:45.518506  6493 solver.cpp:408]     Test net output #0: accuracy = 0.1
I1120 10:36:45.518590  6493 solver.cpp:408]     Test net output #1: loss = 8.60513 (* 1 = 8.60513 loss)
I1120 10:36:45.528278  6493 solver.cpp:236] Iteration 2880, loss = 0.00259783
I1120 10:36:45.528314  6493 solver.cpp:252]     Train net output #0: accuracy = 1
I1120 10:36:45.528333  6493 solver.cpp:252]     Train net output #1: loss = 0.00259783 (* 1 = 0.00259783 loss)
I1120 10:36:45.528362  6493 sgd_solver.cpp:106] Iteration 2880, lr = 0.00578977
I1120 10:36:46.868722  6493 solver.cpp:461] Snapshotting to binary proto file models/a_iter_2920.caffemodel
I1120 10:36:46.907760  6493 sgd_solver.cpp:269] Snapshotting solver state to binary proto file models/a_iter_2920.solverstate
I1120 10:36:48.265416  6493 solver.cpp:461] Snapshotting to binary proto file models/a_iter_2960.caffemodel
I1120 10:36:48.303402  6493 sgd_solver.cpp:269] Snapshotting solver state to binary proto file models/a_iter_2960.solverstate
I1120 10:36:48.310693  6493 solver.cpp:340] Iteration 2960, Testing net (#0)
I1120 10:36:48.908462  6493 solver.cpp:408]     Test net output #0: accuracy = 0.1
I1120 10:36:48.908543  6493 solver.cpp:408]     Test net output #1: loss = 21.0013 (* 1 = 21.0013 loss)
I1120 10:36:48.918262  6493 solver.cpp:236] Iteration 2960, loss = -1.5273e-07
I1120 10:36:48.918299  6493 solver.cpp:252]     Train net output #0: accuracy = 1
I1120 10:36:48.918319  6493 solver.cpp:252]     Train net output #1: loss = 0 (* 1 = 0 loss)
I1120 10:36:48.918346  6493 sgd_solver.cpp:106] Iteration 2960, lr = 0.00576294
I1120 10:36:50.259600  6493 solver.cpp:461] Snapshotting to binary proto file models/a_iter_3000.caffemodel
I1120 10:36:50.302613  6493 sgd_solver.cpp:269] Snapshotting solver state to binary proto file models/a_iter_3000.solverstate
I1120 10:36:51.660645  6493 solver.cpp:461] Snapshotting to binary proto file models/a_iter_3040.caffemodel
I1120 10:36:51.699115  6493 sgd_solver.cpp:269] Snapshotting solver state to binary proto file models/a_iter_3040.solverstate
I1120 10:36:51.706848  6493 solver.cpp:340] Iteration 3040, Testing net (#0)
I1120 10:36:52.303458  6493 solver.cpp:408]     Test net output #0: accuracy = 0
I1120 10:36:52.303532  6493 solver.cpp:408]     Test net output #1: loss = 27.1849 (* 1 = 27.1849 loss)
I1120 10:36:52.313338  6493 solver.cpp:236] Iteration 3040, loss = 6.42871e-08
I1120 10:36:52.313374  6493 solver.cpp:252]     Train net output #0: accuracy = 1
I1120 10:36:52.313393  6493 solver.cpp:252]     Train net output #1: loss = 0 (* 1 = 0 loss)
I1120 10:36:52.313421  6493 sgd_solver.cpp:106] Iteration 3040, lr = 0.00573641
I1120 10:36:53.653604  6493 solver.cpp:461] Snapshotting to binary proto file models/a_iter_3080.caffemodel
I1120 10:36:53.693169  6493 sgd_solver.cpp:269] Snapshotting solver state to binary proto file models/a_iter_3080.solverstate
I1120 10:36:55.051772  6493 solver.cpp:461] Snapshotting to binary proto file models/a_iter_3120.caffemodel
I1120 10:36:55.090648  6493 sgd_solver.cpp:269] Snapshotting solver state to binary proto file models/a_iter_3120.solverstate
I1120 10:36:55.098688  6493 solver.cpp:340] Iteration 3120, Testing net (#0)
I1120 10:36:55.695263  6493 solver.cpp:408]     Test net output #0: accuracy = 0.1
I1120 10:36:55.695341  6493 solver.cpp:408]     Test net output #1: loss = 28.1593 (* 1 = 28.1593 loss)
I1120 10:36:55.705041  6493 solver.cpp:236] Iteration 3120, loss = 5.56233e-08
I1120 10:36:55.705076  6493 solver.cpp:252]     Train net output #0: accuracy = 1
I1120 10:36:55.705101  6493 solver.cpp:252]     Train net output #1: loss = 0 (* 1 = 0 loss)
I1120 10:36:55.705128  6493 sgd_solver.cpp:106] Iteration 3120, lr = 0.00571015
I1120 10:36:57.044112  6493 solver.cpp:461] Snapshotting to binary proto file models/a_iter_3160.caffemodel
I1120 10:36:57.082219  6493 sgd_solver.cpp:269] Snapshotting solver state to binary proto file models/a_iter_3160.solverstate
I1120 10:36:58.439213  6493 solver.cpp:461] Snapshotting to binary proto file models/a_iter_3200.caffemodel
I1120 10:36:58.477608  6493 sgd_solver.cpp:269] Snapshotting solver state to binary proto file models/a_iter_3200.solverstate
I1120 10:36:58.485158  6493 solver.cpp:340] Iteration 3200, Testing net (#0)
I1120 10:36:59.081439  6493 solver.cpp:408]     Test net output #0: accuracy = 0
I1120 10:36:59.081514  6493 solver.cpp:408]     Test net output #1: loss = 27.8542 (* 1 = 27.8542 loss)
I1120 10:36:59.091503  6493 solver.cpp:236] Iteration 3200, loss = 0.528649
I1120 10:36:59.091538  6493 solver.cpp:252]     Train net output #0: accuracy = 0
I1120 10:36:59.091558  6493 solver.cpp:252]     Train net output #1: loss = 0.528649 (* 1 = 0.528649 loss)
